src/vs/platform/dialogs/common/dialogs.ts

```diff
@@ -154,7 +154,12 @@ export interface IPromptResultWithCancel<T> extends IPromptResult<T> {
 
 export type IDialogResult = IConfirmationResult | IInputResult | IPromptResult<unknown>;
 
-export type DialogType = 'none' | 'info' | 'error' | 'question' | 'warning';
+export type DialogType = 'none' | 'info' | 'error' | 'question' | 'warning' | 'success';
```

Comment:
@bpasero this PR introduces a success/confirmation dialog to support a Share experience, given prior art in Powerpoint/Word, e.g.:
<img width="360" alt="image" src="https://github.com/microsoft/vscode/assets/30305945/5d3e3355-e114-4923-9874-0441dccf7dc6">

This also addresses a pain point where copy link actions today do not display success/confirmation and can fail silently. I would appreciate your review of the dialog-related changes here please--thank you!